src/ExpressReceiver.ts

```diff
@@ -201,32 +201,72 @@ export default class ExpressReceiver implements Receiver {
     this.bolt = bolt;
   }
 
-  // TODO: the arguments should be defined as the arguments of Server#listen()
-  // TODO: the return value should be defined as a type that both http and https servers inherit from, or a union
-  public start(port: number): Promise<Server> {
+  // TODO: can this method be defined as generic instead of using overloads?
+  public start(port: number): Promise<Server>;
+  public start(portOrListenOptions: number | ListenOptions, serverOptions?: ServerOptions): Promise<Server>;
+  public start(portOrListenOptions: number | ListenOptions, httpsServerOptions?: HTTPSServerOptions): Promise<HTTPSServer>;
+  public start(portOrListenOptions: number | ListenOptions, serverOptions: ServerOptions | HTTPSServerOptions = {}): Promise<Server | HTTPSServer> {
+    let createServerFn: (typeof createServer | typeof createHttpsServer) = createServer;
+
+    // Decide which kind of server, HTTP or HTTPS, by search for any keys in the serverOptions that are exclusive to HTTPS
+    if (Object.keys(serverOptions).filter(k => httpsOptionKeys.includes(k)).length > 0) {
+      createServerFn = createHttpsServer;
+    }
+
+    if (this.server !== undefined) {
+      return Promise.reject(new ReceiverInconsistentStateError('The receiver cannot be started because it was already started.'));
+    }
+
+    this.server = createServerFn(serverOptions, this.app);
```

Comment:
If we can have some ways to write unit tests for this method call (createServer) without calling `listen` method, it's easier to verify the method's behavior with a variety of options. This is beneficial for future maintenance. Do you have any ideas about it?

Reply:
Also, if possible, we should have some tests for this change.

Reply:
I hope this could be done with `rewiremock`. I'll try to implement some tests.